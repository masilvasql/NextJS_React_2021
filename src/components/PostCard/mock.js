export const PostCardPropsMock = {
    title: 'TÃ­tulo',
    body: 'Body',
    cover: 'Imagem/img.png'
}