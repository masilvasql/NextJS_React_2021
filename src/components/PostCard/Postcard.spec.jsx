
import {PostCard} from '.'
import { PostCardPropsMock } from './mock'
const { render, screen } = require("@testing-library/react")

const props = PostCardPropsMock;

describe('<PostCard/>', ()=>{
    it('should render PostCard correctly', ()=>{
        // const {debug} =render(<PostCard {...props} />)
        // debug();

        render(<PostCard {...props} />)

        expect(screen.getByRole('img', {name:'Título'}))
        .toHaveAttribute('src', 'Imagem/img.png');

        expect(screen.getByRole('heading', {name:'Título'})).toBeInTheDocument()
        expect(screen.getByText('Body')).toBeInTheDocument()
    })

    it('should match snapshot', ()=>{
        const  {container } = render(<PostCard {...props} />)
        expect(container.firstChild).toMatchSnapshot();
    })
})