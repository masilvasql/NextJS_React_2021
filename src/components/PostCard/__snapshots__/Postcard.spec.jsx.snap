// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard/> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="Título"
    src="Imagem/img.png"
  />
  <div
    class="post-content"
  >
    <h1>
      Título
    </h1>
    <p>
      Body
    </p>
  </div>
</div>
`;
